src/app/inventory/PullFromBucketButton.tsx

```diff
@@ -0,0 +1,55 @@
+import { t } from 'app/i18next-t';
+import { ThunkDispatchProp } from 'app/store/types';
+import { itemCanBeEquippedBy } from 'app/utils/item-utils';
+import React, { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { showItemPicker } from '../item-picker/item-picker';
+import { addIcon, AppIcon } from '../shell/icons';
+import { InventoryBucket } from './inventory-buckets';
+import { DimItem } from './item-types';
+import { moveItemTo } from './move-item';
+import { DimStore } from './store-types';
+
+interface Props {
+  className?: string;
+  label?: string;
+  store: DimStore;
+  bucket: InventoryBucket;
+}
+
+export default function PullFromBucketButton({ label, className, bucket, store }: Props) {
+  const dispatch = useDispatch<ThunkDispatchProp['dispatch']>();
+
+  const pullItem = useCallback(async () => {
+    try {
+      const { item } = await showItemPicker({
+        filterItems: (item: DimItem) =>
+          item.bucket.hash === bucket.hash && itemCanBeEquippedBy(item, store),
+        prompt: t('MovePopup.PullItem', {
+          bucket: bucket.name,
+          store: store.name,
+        }),
+      });
+
+      dispatch(moveItemTo(item, store));
+    } catch (e) {}
+  }, [bucket.hash, bucket.name, dispatch, store]);
```

Comment:
Might have to revisit this, I've moved this off into a separate action now.

Reply:
Thanks for the heads up, I'll rebase and land w/ the background-color updates this evening.

Reply:
which action? I'm extracting this logic: 
https://github.com/DestinyItemManager/DIM/blob/master/src/app/inventory/StoreBucket.tsx#L77-L92

Reply:
Haha I think the change I'm describing is still on a branch somewhere